internal/repository/bot_repo.go

```diff
@@ -0,0 +1,58 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"github.com/sirupsen/logrus"
+)
+
+type BOTRepository interface {
+	SaveURL(ctx context.Context, req SaveUrlRequest) error
```

Comment:
req *SaveUrlRequest

Reply:
if using struct params (req SaveUrlRequest) then add Request to other methods in your interface